ipinfo/handler.py

```diff
@@ -305,3 +313,72 @@ def getMap(self, ips):
         )
         response.raise_for_status()
         return response.json()["reportUrl"]
+
+    def getBatchDetailsIter(
+        self,
+        ip_addresses,
+        batch_size=None,
+        raise_on_fail=True,
+    ):
+        if batch_size is None:
+            batch_size = BATCH_MAX_SIZE
+
+        result = {}
+        lookup_addresses = []
+        for ip_address in ip_addresses:
+            if isinstance(ip_address, IPv4Address) or isinstance(
+                ip_address, IPv6Address
+            ):
+                ip_address = ip_address.exploded
+
+            if ip_address and is_bogon(ip_address):
+                details = {}
+                details["ip"] = ip_address
+                details["bogon"] = True
+                yield Details(details)
+            else:
+                lookup_addresses.append(ip_address)
+
+            try:
+                cached_ipaddr = self.cache[cache_key(ip_address)]
+                result[ip_address] = cached_ipaddr
+            except KeyError:
+                lookup_addresses.append(ip_address)
+
+        # all in cache - exit early.
+        if len(lookup_addresses) == 0:
+            yield result.items()
```

Comment:
This is not exiting. It'll continue executing the function on the next loop?

Reply:
I've updated, please review if it fixed it now.